### PR TITLE
CAMEL-19732 - Support Pipe in CRDs, replacing KameletBinding

### PR DESCRIPTION
**Pull request by apupier**
keeping KameletBinding working to let transition times for users

# Description

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/common/GitHubHelper.java [hunk 1]
```diff
@@ -63,11 +63,6 @@ private static void doFetchGithubUrls(
 
         // this is a directory, so we need to query github which files are there and filter them
 
-        // URL: https://api.github.com/repos/apache/camel-k/contents/examples/kamelets/kameletbindings
-        // URL: https://api.github.com/repos/apache/camel-k/contents/examples/kamelets/kameletbindings?ref=v1.7.0
-        // https://github.com/apache/camel-k/tree/main/examples/kamelets/kameletbindings
-        // https://github.com/apache/camel-k/tree/v1.7.0/examples/kamelets/kameletbindings
-
```
Comment (apupier) [inline dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/common/GitHubHelper.java:70]: All these URLs does not exists anymore. And I have not found a specific folder of examples for Pipe/KameletBinding

Comment (claudio4j) [reply dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/common/GitHubHelper.java:70]: There are some examples in
https://github.com/apache/camel-kamelets-examples/
https://github.com/apache/camel-k-examples/tree/main/generic-examples/kamelets
https://github.com/apache/camel-k-examples/tree/main/kamelets

Comment (apupier) [reply dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/common/GitHubHelper.java:70]: there are examples but they are not only kameletbindings as it was in the comment

Comment (apupier) [reply dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/common/GitHubHelper.java:70]: there are no tags on camel-k-examples repository neither (and in the url provided it was using tags for some)

**Comment by davsclaus** [comment]
Can you copy one of the pipe tests and change it to use the old name, so we have 1 test for backwards

**Comment by apupier** [comment]
> Can you copy one of the pipe tests and change it to use the old name, so we have 1 test for backwards

I kept the old one: https://github.com/apupier/camel/blob/19732-migrateFromKameletBindingToPipeNaming/dsl/camel-yaml-dsl/camel-yaml-dsl/src/test/groovy/org/apache/camel/dsl/yaml/KameletBindingLoaderTest.groovy

**Comment by davsclaus** [comment]
Ah okay, didnt spot that.

But maybe we could just deleted all but 1 old test as its the same logic the Pipe uses, so we dont have to maintain old tests.

**Review by claudio4j — APPROVED** [review]

**Comment by apupier** [comment]
> we could just deleted all but 1 old test as its the same logic the Pipe uses, so we dont have to maintain old tests.

updated

**Review by davsclaus — APPROVED** [review]

**Comment by apupier** [comment]
the PR is failign with: https://github.com/apache/camel/actions/runs/5830292917/job/15811436755#step:8:27

```
 There are uncommitted changes
HEAD detached at pull/11067/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd

no changes added to commit (use "git add" and/or "git commit -a")


diff --git a/catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd b/catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd
index bb6822e416d..0b662159251 100644
--- a/catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd
+++ b/catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd
@@ -439,7 +439,7 @@ Enriches a message with data from a secondary resource
     <xs:annotation>
       <xs:documentation xml:lang="en">
 <![CDATA[
-Error handler settings
+Camel error handling.
 ]]>
       </xs:documentation>
     </xs:annotation>
```

i do not see the link with my PR. I had this change locally one time when rebuilding.

**Pull request merged**